code/src/Components/MovieList.js

```diff
@@ -0,0 +1,390 @@
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'Romance') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10749))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10749))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'SciFi') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(878))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(878))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'TVmovie') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10770))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10770))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'Thriller') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(53))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(53))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'War') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10752))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10752))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'Western') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(37))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(37))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		}
+	}, [filter])
```

Comment:
We think this could be a function where you pass in the genre-numbers as an argument and then map over the genre-array. This would make the code a lot shorter.

`
const genreFilter = ( movienumber, movietype ) => 		
const filteredList = [...movieList]
			if (sorting === movietype) {
				setfilteredMovieList(
					filteredList
						.filter((movie) => movie.genre_ids.includes(movienumber))
						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
				)
			} else {
				setfilteredMovieList(
					filteredList
						.filter((movie) => movie.genre_ids.includes(movienumber))
						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
				)
			}
genreFilter(28, "release")
`